Program.cs

```diff
@@ -123,40 +156,27 @@ public static void Main()
             if (!gameOver && currentFrame - lastFrameMoved >= secondsToMove)
             {
                 lastFrameMoved = currentFrame;
-                snake.Move(gridRows, gridCols);
-                gameOver = snake.Overlaps(snake.snakeRow, snake.snakeCol, score, skipHead: true);
+                snake.Move(grid);
+                gameOver = snake.Overlaps(snake.Head, score, skipHead: true);
             }
 
-            // Debug code for printnig arrays 
-            // foreach (var item in snake.prevLocRow)
-            // {
-            //     Console.Write(item.ToString(), ", ");
-            // }
-            // Console.WriteLine();
-
-            // foreach (var item in snake.prevLocCol)
-            // {
-            //     Console.Write(item.ToString(), ", ");
-            // }
-            // Console.WriteLine();
-
-            if (snake.Overlaps(apple.row, apple.col, score))
+            if (snake.Overlaps(apple.appleCords, score))
             {
                 score += 1;
                 Console.WriteLine("Score:" + score);
-                secondsToMove *= speedFactor; //if i include this will it not be too big?
+                secondsToMove *= config.speedFactor; //if i include this will it not be too big?
                 apple.Respawn(snake, score);
-                center = CellToCenter(apple.row, apple.col, squareSize);
+                center = CellToCenter(apple.appleCords, config.squareSize);
             }
 
             // DRAWING SECTION 
             Raylib.BeginDrawing();
             Raylib.ClearBackground(Color.White);
 
-            DrawGrid(gridRows, gridCols, squareSize);
+            DrawGrid(grid, config.squareSize);
```

Comment:
Right now you only need the square size from the config. 

But here's a question to mull over. Respond here with pro's and cons of doing what you have 
```
            DrawGrid(grid, config.squareSize);
```

and something like this ?
```
            DrawGrid(grid, config);
```